sniffles/snifflesconfig.py

```diff
@@ -459,7 +466,20 @@ def parse_opt_arg(self, opt, arg):
             self.tcp_ack = True
         # Set percentage of background traffics to be added
         elif opt == "-B":
-            self.background_traffic = int(arg)
+            args = re.split('[:]+', arg)
+            if len(args) == 1:
+                self.background_traffic = int(args[0])
+            else:
+               args = re.split('[,]+', arg)
+               for i in range(0, len(args)):
+                   args_dist = re.split('[:]+', args[i])
+                   if len(args_dist) == 1:
+                       self.usage()
+                   else:
+                       self.background_traffic += int(args_dist[1])
```

Comment:
in rulereader.py 
```
 781     def updateProbability(self):
 782         for protocol in self.application_protocol:
 783             if protocol in self.distribution:
 784                 self.probability_list += (self.distribution[protocol] *
 785                                           [protocol])
 786             else:
 787                 self.absent_protocol.append(protocol)
 788         if len(self.probability_list) < 100:
 789             multiplier = 100 - len(self.probability_list)
 790             self.probability_list += multiplier * ['remainder']
```
additionally, In the existing method, the backgroundPercent and the distribution are independent, so I have to choose to implement the function.
1. Create a probability_list[] with updateProbability() and remove the 'remainder'.
2. Modify the above updateProbability () like adding parameters.
Please advise which method is good. thank you.